File: add.js
window.onload = function() {
  initMap();
};

let map;
let marker;

function initMap() {
  const defaultLocation = { lat: 41.9981, lng: 21.4254 }; // Skopje coordinates
  map = new google.maps.Map(document.getElementById("map"), {
    center: defaultLocation,
    zoom: 8
  });

  map.addListener("click", function(event) {
    placeMarkerAndPanTo(event.latLng, map);
  });
}

function placeMarkerAndPanTo(latLng, map) {
  if (marker) {
    marker.setMap(null);
  }

  marker = new google.maps.Marker({
    position: latLng,
    map: map
  });

  document.getElementById("latitude").value = latLng.lat();
  document.getElementById("longitude").value = latLng.lng();
}

// Form submission
const form = document.getElementById('editParkingForm');
form.addEventListener('submit', async function(event) {
  event.preventDefault();

  const name = document.getElementById('name').value;
  const price = document.getElementById('price').value;
  const address = document.getElementById('address').value;
  const zone = document.getElementById('zone').value;
  const startTime = document.getElementById('startTime').value;
  const endTime = document.getElementById('endTime').value;
  const capacityCar = document.getElementById('capacityCar').value;
  const capacityAdapted = document.getElementById('capacityAdapted').value;
  const latitude = document.getElementById('latitude').value;
  const longitude = document.getElementById('longitude').value;

  if (name && price && address && zone && startTime && endTime && capacityCar && capacityAdapted && latitude && longitude) {
    try {
      // Save data to Firestore
      await db.collection('parkingLots').add({
        name: name,
        price: parseFloat(price),
        address: address,
        zone: zone,
        workingHours: {
          start: startTime,
          end: endTime
        },
        capacity: {
          car: parseInt(capacityCar),
          adapted: parseInt(capacityAdapted)
        },
        location: {
          latitude: parseFloat(latitude),
          longitude: parseFloat(longitude)
        }
      });

      alert("Parking lot details saved successfully!");
      form.reset(); // Reset the form
    } catch (error) {
      console.error("Error saving parking lot: ", error);
      alert("Error saving parking lot details!");
    }
  } else {
    alert("Please fill in all fields!");
  }
});

File: add.py
from flask import Flask, request, render_template, redirect, url_for
import firebase_admin
from firebase_admin import credentials, firestore

# Initialize the Flask app
app = Flask(__name__)

# Path to your Firebase service account key file
cred = credentials.Certificate('path/to/your/service.json')
firebase_admin.initialize_app(cred)

# Initialize Firestore
db = firestore.client()

# Route for the form page
@app.route('/')
def index():
    return render_template('Add.html')

# Route to handle form submission
@app.route('/submit', methods=['POST'])
def submit_form():
    # Get data from the form
    name = request.form['name']
    price = request.form['price']
    address = request.form['address']
    zone = request.form['zone']
    start_time = request.form['startTime']
    end_time = request.form['endTime']
    capacity_car = request.form['capacityCar']
    capacity_adapted = request.form['capacityAdapted']
    latitude = request.form['latitude']
    longitude = request.form['longitude']

    # Create a dictionary to store parking lot data
    parking_lot_data = {
        'name': name,
        'price': float(price),
        'address': address,
        'zone': zone,
        'workingHours': {
            'start': start_time,
            'end': end_time
        },
        'capacity': {
            'car': int(capacity_car),
            'adapted': int(capacity_adapted)
        },
        'location': {
            'latitude': float(latitude),
            'longitude': float(longitude)
        }
    }

    # Save to Firestore
    try:
        db.collection('parkingLots').add(parking_lot_data)
        return redirect(url_for('index', message='Parking lot added successfully!'))
    except Exception as e:
        return f"An Error Occurred: {str(e)}"

if __name__ == '__main__':
    app.run(debug=True)
